From efcb98be2024f15e4dba3d8c36ed9cd55a287598 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Wed, 26 Sep 2018 13:50:43 +0800
Subject: [PATCH 4735/5242] MLK-19730: ISI: correct V4L2_PIX_FMT_RGB32 to
 V4L2_PIX_FMT_XRGB32

commit  28e2c6388c5135e05f676409a4cb744d20ec695c from
https://source.codeaurora.org/external/imx/linux-imx.git

Because V4L2_PIX_FMT_RGB32 is deprecated and must not
be used by new drivers. V4L2_PIX_FMT_XRGB32 is used to
replace it.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit a6780d01aa9219c6a720177bb85af755ba09a24a)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-isi-cap.c |    2 +-
 drivers/media/platform/imx8/mxc-isi-hw.c  |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/imx8/mxc-isi-cap.c b/drivers/media/platform/imx8/mxc-isi-cap.c
index c684b6a..7c5f7d4 100644
--- a/drivers/media/platform/imx8/mxc-isi-cap.c
+++ b/drivers/media/platform/imx8/mxc-isi-cap.c
@@ -52,7 +52,7 @@ struct mxc_isi_fmt mxc_isi_out_formats[] = {
 		.mbus_code  = MEDIA_BUS_FMT_RGB888_1X24,
 	}, {
 		.name		= "RGB32",
-		.fourcc		= V4L2_PIX_FMT_RGB32,
+		.fourcc		= V4L2_PIX_FMT_XRGB32,
 		.depth		= { 32 },
 		.color		= MXC_ISI_OUT_FMT_XRGB32,
 		.memplanes	= 1,
diff --git a/drivers/media/platform/imx8/mxc-isi-hw.c b/drivers/media/platform/imx8/mxc-isi-hw.c
index 5e836ac..98e8421 100644
--- a/drivers/media/platform/imx8/mxc-isi-hw.c
+++ b/drivers/media/platform/imx8/mxc-isi-hw.c
@@ -84,7 +84,7 @@ static bool is_rgb(u32 pix_fmt)
 {
 	if ((pix_fmt == V4L2_PIX_FMT_RGB565) ||
 		(pix_fmt == V4L2_PIX_FMT_RGB24) ||
-		(pix_fmt == V4L2_PIX_FMT_RGB32) ||
+		(pix_fmt == V4L2_PIX_FMT_XRGB32) ||
 		(pix_fmt == V4L2_PIX_FMT_BGR24) ||
 	    (pix_fmt == V4L2_PIX_FMT_ARGB32)) {
 		return true;
-- 
1.7.9.5

